pyproject.toml

```diff
@@ -233,3 +233,84 @@ legacy_tox_ini = """
     commands_post =
         python -c 'import pathlib; print(f"Package is under:", pathlib.Path(f"{tox_root}") / "dist")'
 """
+
+[tool.ruff]
+exclude = [
+    ".git",
+    "__pycache__",
+    "build",
+    "docs/_build",
+    "dist"
+]
+ignore = [
+    # line too long -> we accept long comment lines; black gets rid of long code lines
+    "E501",
+    # Do not assign a lambda expression, use a def -> lambda expression assignments are convenient
+    "E731",
+    # allow I, O, l as variable names -> I is the identity matrix, i, j, k, l is reasonable indexing notation
+    "E741",
+    # Missing docstring in public package
+    "D104",
+    # ... imported but unused
+    "F401",
+    # Missing docstring in public module
+    "D100",
+    # Missing docstring in __init__
+    "D107",
+    # Do not perform function calls in argument defaults.
+    "B008",
+    # Missing docstring in magic method
+    "D105",
+    # Missing blank line before section
+    "D411",
+    ## Flake8 rules not supported by ruff:
+    # line break before a binary operator -> black does not adhere to PEP8
+    # "W503",
+    # line break occured after a binary operator -> black does not adhere to PEP8
+    # "W504",
+    # whitespace before : -> black does not adhere to PEP8
+    # "E203",
+    # whitespace before : -> black does not adhere to PEP8
+    # "E203",
+    # missing whitespace after ,', ';', or ':' -> black does not adhere to PEP8
+    # "E231",
+    # continuation line over-indented for hanging indent -> black does not adhere to PEP8
+    # "E126",
+    # too many leading '# ' for block comment -> Scanpy allows them for comments into sections
+    # "E262",
+    # inline comment should start with '#' -> Scanpy allows them for specific explanations
+    # "E266",
+    # format string does contain unindexed parameters
+    # "P101",
+    # indentation is not a multiple of 4
+    # "E111",
+    # "E114",
+]
+line-length = 80
+select = [
+    "I", # isort
+    "E", # pycodestyle
+    "F", # pyflakes
+    "W", # pycodestyle
+    # below are not autofixed
+    "UP", # pyupgrade
+    "C4", # flake8-comprehensions
+    "B", # flake8-bugbear
+    "BLE", # flake8-blind-except
+]
+unfixable = ["B", "UP", "C4", "BLE"]
+target-version = "py38"
+[tool.ruff.per-file-ignores]
+    "tests/*" = ["D", "E", "F", "W", "I", "C408"]
+    "*/__init__.py" = ["F401"]
+    "examples/*" = ["D101", "D102", "D103"]
+    "docs/*" = ["E", "F", "W", "I", "D101", "D102"]
+    "src/ott/types.py" = ["D102"]
+[tool.ruff.pydocstyle]
+convention = "google"
+[tool.ruff.pyupgrade]
+# Preserve types, even if a file imports `from __future__ import annotations`.
+keep-runtime-typing = true
+[tool.ruff.flake8-tidy-imports]
+# Disallow all relative imports.
+ban-relative-imports = "all"
```

Comment:
I think the value is wrong, see [here](https://github.com/adamchainz/flake8-tidy-imports#ban-relative-imports).
But I also think we wanna keep relative imports, given we have only 80 columns, pinging @marcocuturi .

Reply:
Ok, you can add this, but please check the value is correct - in `__init__.py`, relative imports are still present.

Reply:
ok shall I ban all or just parents?

Reply:
I added `parents` option, so init are not touched

Reply:
Thanks!